src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard-v3/dashboard/dashboard-v3.component.ts

```diff
@@ -166,12 +167,14 @@ export class DashboardV3Component extends PrometheusListHelper implements OnInit
   }
 
   getCapacityCardData() {
-    this.osdSettingsService = this.osdService
-      .getOsdSettings()
-      .pipe(take(1))
-      .subscribe((data: any) => {
-        this.osdSettings = data;
-      });
+    if (this.permissions.configOpt.read) {
```

Comment:
this condition should be outside where the function is being called not inside the capacity function.
If we are keepin inside then return defaults but better to move outside if possible

Reply:
but the general functions responsibility is to set the capacity and not just understand the osdSettings, which is what happens on the next line

```

    this.capacityService = this.healthService.getClusterCapacity().subscribe((data: any) => {
      this.capacity = data;
    });
```

so I don't think we cannot move this outside of this function.

Reply:
for now I've moved osdSettings to its own function `getOsdSettings` and renamed `getCapacityCardData` to `getCapacity`.